pkg/virt-launcher/virtwrap/sriov/pcipool_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package sriov_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/sriov"
+)
+
+var _ = Describe("SRIOV PCI address pool", func() {
+	It("Fails to pop an address given no interfaces", func() {
+		pool := sriov.NewPCIAddressPool([]v1.Interface{})
+		expectPoolPopFailure(pool, "foo")
+	})
+
+	It("fails to pop an address given a non-SRIOV interface", func() {
+		pool := sriov.NewPCIAddressPool([]v1.Interface{{Name: "foo"}})
+		expectPoolPopFailure(pool, "foo")
+	})
+
+	It("fails to pop an address given a missing resource name env", func() {
+		iface := newSRIOVInterface("foo")
+		pool := sriov.NewPCIAddressPool([]v1.Interface{iface})
+
+		expectPoolPopFailure(pool, "foo")
+	})
+
+	It("fails to pop an address given a missing device name env", func() {
```

Comment:
Can we add some low level tests like, pci address with spaces between commas and duplicated pci addresses ?

Reply:
Do you have a pointer to the object or commit ?